Index: open-iscsi-2.0-873/usr/iface.c
===================================================================
--- open-iscsi-2.0-873.orig/usr/iface.c
+++ open-iscsi-2.0-873/usr/iface.c
@@ -896,6 +896,7 @@ int iface_setup_from_boot_context(struct
 	struct iscsi_transport *t = NULL;
 	char transport_name[ISCSI_TRANSPORT_NAME_MAXLEN];
 	uint32_t hostno;
+	int rc;
 
 	if (strlen(context->initiatorname))
 		strlcpy(iface->iname, context->initiatorname,
@@ -908,9 +909,6 @@ int iface_setup_from_boot_context(struct
 			return 0;
 		}
 	} else if (strlen(context->iface)) {
-/* this ifdef is only temp until distros and firmwares are updated */
-#ifdef OFFLOAD_BOOT_SUPPORTED
-
 		memset(transport_name, 0, ISCSI_TRANSPORT_NAME_MAXLEN);
 		/* make sure offload driver is loaded */
 		if (!net_get_transport_name_from_netdev(context->iface,
@@ -931,9 +929,6 @@ int iface_setup_from_boot_context(struct
 		}
 
 		strlcpy(iface->netdev, context->iface, sizeof(iface->netdev));
-#else
-		return 0;
-#endif
 	} else
 		return 0;
 
