--- open-iscsi-2.0-872-rc4-bnx2i/usr/version.h	2011-09-01 20:54:51.000000000 -0500
+++ open-iscsi-2.0-872-rc4-bnx2i.build/usr/version.h	2011-09-01 20:59:20.000000000 -0500
@@ -6,7 +6,7 @@
  * This may not be the same value as the kernel versions because
  * some other maintainer could merge a patch without going through us
  */
-#define ISCSI_VERSION_STR	"2.0-872"
+#define ISCSI_VERSION_STR	"2.0-872.15.f16"
 #define ISCSI_VERSION_FILE	"/sys/module/scsi_transport_iscsi/version"
 
 #endif
