diff -aurp open-iscsi-2.0-872-rc4-bnx2i/usr/initiator.c open-iscsi-2.0-872-rc4-bnx2i.build/usr/initiator.c
--- open-iscsi-2.0-872-rc4-bnx2i/usr/initiator.c	2011-09-01 20:28:53.000000000 -0500
+++ open-iscsi-2.0-872-rc4-bnx2i.build/usr/initiator.c	2011-09-01 20:29:49.000000000 -0500
@@ -484,8 +484,7 @@ cleanup:
 	if (session->id != -1) {
 		log_debug(2, "kdestroy session %u", session->id);
 		session->r_stage = R_STAGE_SESSION_DESTOYED;
-		err = ipc->destroy_session(session->t->handle, session->id);
-		if (err) {
+		if (ipc->destroy_session(session->t->handle, session->id)) {
 			log_error("can not safely destroy session %d",
 				  session->id);
 			return ISCSI_ERR_INTERNAL;
