Index: b/Makefile
===================================================================
--- a/Makefile
+++ b/Makefile
@@ -12,6 +12,12 @@ VERSION := $(shell awk '/Version:/ { pri
 RELEASE := $(shell awk -F '[ %]' '/Release:/ { print $$2 }' python-pyblock.spec)
 USESELINUX = 1
 
+python_INCLUDES := $(shell python-config --includes)
+CFLAGS += $(python_INCLUDES)
+
+python_LIBS := $(shell python-config --libs)
+LDFLAGS += $(python_LIBS)
+
 dm_LIBS = dmraid devmapper
 ifeq (1, $(USESELINUX))
 dm_LIBS += selinux
@@ -23,15 +29,19 @@ PYFILES=__init__.py maps.py device.py
 LIBS = dmmodule.so dmraidmodule.so
 
 all : $(LIBS)
+	echo "python_INCLUDES $(python_INCLUDES)"
+	echo "python_LIBS $(python_LIBS)"
+	echo "CFLAGS $(CFLAGS)"
+	echo "LDFLAGS $(LDFLAGS)"
 
 %.o : %.c
-	$(CC) $(CFLAGS) -c -o $@ $<
+	$(CC) $(CFLAGS) $(python_INCLUDES) -c -o $@ $<
 
 dmmodule.so.$(VERSION) : dm.o pyhelpers.o exc.o
-	$(CC) $(CFLAGS) $(LDFLAGS) -Wl,-soname="$@" -o $@ $^ $(foreach x,$(value dm_LIBS),-l$(x))
+	$(CC) $(CFLAGS) $(LDFLAGS) -shared -Wl,-soname="$@" -o $@ $^ $(foreach x,$(value dm_LIBS),-l$(x)) $(python_LIBS)
 
 dmraidmodule.so.$(VERSION) : dmraid.o pyhelpers.o
-	$(CC) $(CFLAGS) $(LDFLAGS) -Wl,-soname="$@" -o $@ $^ $(foreach x,$(value dmraid_LIBS),-l$(x))
+	$(CC) $(CFLAGS) $(LDFLAGS) -shared -Wl,-soname="$@" -o $@ $^ $(foreach x,$(value dmraid_LIBS),-l$(x)) $(python_LIBS)
 
 %.so : %.so.$(VERSION)
 	@echo -n
