Index: fontforge-20120731-b/configure.dynamic.in
===================================================================
--- fontforge-20120731-b.orig/configure.dynamic.in
+++ fontforge-20120731-b/configure.dynamic.in
@@ -708,12 +708,18 @@ if test "$gww_nodynamic" = "yes" -o "$gw
  if test "$ac_cv_lib_png12_png_create_read_struct" = "yes"; then
   STATIC_LIBS="$STATIC_LIBS -lpng12 -lz"
  fi
+ if test "$ac_cv_lib_png13_png_create_read_struct" = "yes"; then
+  STATIC_LIBS="$STATIC_LIBS -lpng13 -lz"
+ fi
  if test "$ac_cv_lib_png14_png_create_read_struct" = "yes"; then
   STATIC_LIBS="$STATIC_LIBS -lpng14 -lz"
  fi
  if test "$ac_cv_lib_png15_png_create_read_struct" = "yes"; then
   STATIC_LIBS="$STATIC_LIBS -lpng15 -lz"
  fi
+ if test "$ac_cv_lib_png16_png_create_read_struct" = "yes"; then
+  STATIC_LIBS="$STATIC_LIBS -lpng16 -lz"
+ fi
  if test "$ac_cv_lib_tiff_TIFFOpen" = "yes"; then
   STATIC_LIBS="$STATIC_LIBS -ltiff"
  fi
@@ -766,10 +772,22 @@ if test "$gww_needsstaticimages" = "yes"
   STATIC_LIBS="$STATIC_LIBS -lpng12 -lz"
   AC_DEFINE(_STATIC_LIBPNG)
  fi
+ if test "$ac_cv_lib_png13_png_create_read_struct" = "yes"; then
+  STATIC_LIBS="$STATIC_LIBS -lpng13 -lz"
+  AC_DEFINE(_STATIC_LIBPNG)
+ fi
  if test "$ac_cv_lib_png14_png_create_read_struct" = "yes"; then
   STATIC_LIBS="$STATIC_LIBS -lpng14 -lz"
   AC_DEFINE(_STATIC_LIBPNG)
  fi
+ if test "$ac_cv_lib_png15_png_create_read_struct" = "yes"; then
+  STATIC_LIBS="$STATIC_LIBS -lpng15 -lz"
+  AC_DEFINE(_STATIC_LIBPNG)
+ fi
+ if test "$ac_cv_lib_png16_png_create_read_struct" = "yes"; then
+  STATIC_LIBS="$STATIC_LIBS -lpng16 -lz"
+  AC_DEFINE(_STATIC_LIBPNG)
+ fi
  if test "$ac_cv_lib_tiff_TIFFOpen" = "yes"; then
   STATIC_LIBS="$STATIC_LIBS -ltiff"
   AC_DEFINE(_STATIC_LIBTIFF)
@@ -809,12 +827,18 @@ case "$host" in
  if test "$ac_cv_lib_png12_png_create_read_struct" = "yes"; then
   AC_CHECK_LIB_LA(png12, :, [ STATIC_LIBS="$STATIC_LIBS -lpng12 -lz" ; AC_DEFINE(_STATIC_LIBPNG12)])
  fi
+ if test "$ac_cv_lib_png13_png_create_read_struct" = "yes"; then
+  AC_CHECK_LIB_LA(png12, :, [ STATIC_LIBS="$STATIC_LIBS -lpng13 -lz" ; AC_DEFINE(_STATIC_LIBPNG13)])
+ fi
  if test "$ac_cv_lib_png14_png_create_read_struct" = "yes"; then
   AC_CHECK_LIB_LA(png14, :, [ STATIC_LIBS="$STATIC_LIBS -lpng14 -lz" ; AC_DEFINE(_STATIC_LIBPNG14)])
  fi
  if test "$ac_cv_lib_png15_png_create_read_struct" = "yes"; then
   AC_CHECK_LIB_LA(png15, :, [ STATIC_LIBS="$STATIC_LIBS -lpng15 -lz" ; AC_DEFINE(_STATIC_LIBPNG15)])
  fi
+ if test "$ac_cv_lib_png16_png_create_read_struct" = "yes"; then
+  AC_CHECK_LIB_LA(png16, :, [ STATIC_LIBS="$STATIC_LIBS -lpng16 -lz" ; AC_DEFINE(_STATIC_LIBPNG16)])
+ fi
  if test "$ac_cv_lib_tiff_TIFFOpen" = "yes"; then
   AC_CHECK_LIB_LA(tiff, :, [ STATIC_LIBS="$STATIC_LIBS -ltiff" ; AC_DEFINE(_STATIC_LIBTIFF)])
  fi
Index: fontforge-20120731-b/configure.in
===================================================================
--- fontforge-20120731-b.orig/configure.in
+++ fontforge-20120731-b/configure.in
@@ -742,9 +742,18 @@ if test "$gww_nodynamic" = "yes" -o "$gw
  if test "$ac_cv_lib_png12_png_create_read_struct" = "yes"; then
   STATIC_LIBS="$STATIC_LIBS -lpng12 -lz"
  fi
+ if test "$ac_cv_lib_png13_png_create_read_struct" = "yes"; then
+  STATIC_LIBS="$STATIC_LIBS -lpng13 -lz"
+ fi
  if test "$ac_cv_lib_png14_png_create_read_struct" = "yes"; then
   STATIC_LIBS="$STATIC_LIBS -lpng14 -lz"
  fi
+ if test "$ac_cv_lib_png15_png_create_read_struct" = "yes"; then
+  STATIC_LIBS="$STATIC_LIBS -lpng15 -lz"
+ fi
+ if test "$ac_cv_lib_png16_png_create_read_struct" = "yes"; then
+  STATIC_LIBS="$STATIC_LIBS -lpng16 -lz"
+ fi
  if test "$ac_cv_lib_tiff_TIFFOpen" = "yes"; then
   STATIC_LIBS="$STATIC_LIBS -ltiff"
  fi
@@ -813,10 +822,22 @@ if test "$gww_needsstaticimages" = "yes"
   STATIC_LIBS="$STATIC_LIBS -lpng12 -lz"
   AC_DEFINE(_STATIC_LIBPNG)
  fi
+ if test "$ac_cv_lib_png13_png_create_read_struct" = "yes"; then
+  STATIC_LIBS="$STATIC_LIBS -lpng13 -lz"
+  AC_DEFINE(_STATIC_LIBPNG)
+ fi
  if test "$ac_cv_lib_png14_png_create_read_struct" = "yes"; then
   STATIC_LIBS="$STATIC_LIBS -lpng14 -lz"
   AC_DEFINE(_STATIC_LIBPNG)
  fi
+ if test "$ac_cv_lib_png15_png_create_read_struct" = "yes"; then
+  STATIC_LIBS="$STATIC_LIBS -lpng15 -lz"
+  AC_DEFINE(_STATIC_LIBPNG)
+ fi
+ if test "$ac_cv_lib_png16_png_create_read_struct" = "yes"; then
+  STATIC_LIBS="$STATIC_LIBS -lpng16 -lz"
+  AC_DEFINE(_STATIC_LIBPNG)
+ fi
  if test "$ac_cv_lib_tiff_TIFFOpen" = "yes"; then
   STATIC_LIBS="$STATIC_LIBS -ltiff"
   AC_DEFINE(_STATIC_LIBTIFF)
@@ -856,9 +877,18 @@ case "$host" in
  if test "$ac_cv_lib_png12_png_create_read_struct" = "yes"; then
   AC_CHECK_LIB_LA(png12, :, [ STATIC_LIBS="$STATIC_LIBS -lpng12 -lz" ; AC_DEFINE(_STATIC_LIBPNG12)])
  fi
+ if test "$ac_cv_lib_png13_png_create_read_struct" = "yes"; then
+  AC_CHECK_LIB_LA(png13, :, [ STATIC_LIBS="$STATIC_LIBS -lpng13 -lz" ; AC_DEFINE(_STATIC_LIBPNG12)])
+ fi
  if test "$ac_cv_lib_png14_png_create_read_struct" = "yes"; then
   AC_CHECK_LIB_LA(png14, :, [ STATIC_LIBS="$STATIC_LIBS -lpng14 -lz" ; AC_DEFINE(_STATIC_LIBPNG14)])
  fi
+ if test "$ac_cv_lib_png15_png_create_read_struct" = "yes"; then
+  AC_CHECK_LIB_LA(png15, :, [ STATIC_LIBS="$STATIC_LIBS -lpng15 -lz" ; AC_DEFINE(_STATIC_LIBPNG14)])
+ fi
+ if test "$ac_cv_lib_png16_png_create_read_struct" = "yes"; then
+  AC_CHECK_LIB_LA(png16, :, [ STATIC_LIBS="$STATIC_LIBS -lpng16 -lz" ; AC_DEFINE(_STATIC_LIBPNG14)])
+ fi
  if test "$ac_cv_lib_tiff_TIFFOpen" = "yes"; then
   AC_CHECK_LIB_LA(tiff, :, [ STATIC_LIBS="$STATIC_LIBS -ltiff" ; AC_DEFINE(_STATIC_LIBTIFF)])
  fi
Index: fontforge-20120731-b/configure.static.in
===================================================================
--- fontforge-20120731-b.orig/configure.static.in
+++ fontforge-20120731-b/configure.static.in
@@ -504,6 +504,22 @@ if test "$gww_nodynamic" = "yes"; then
   STATIC_LIBS="$STATIC_LIBS -lpng12 -lz"
   AC_DEFINE(_STATIC_LIBPNG)
  fi
+ if test "$ac_cv_lib_png13_png_create_read_struct" = "yes" -a "$ac_cv_lib_z_deflateEnd" = "yes" -a \( ! -e /usr/lib/libpng13.la -o ! -e /usr/lib/libz.la \); then
+  STATIC_LIBS="$STATIC_LIBS -lpng13 -lz"
+  AC_DEFINE(_STATIC_LIBPNG)
+ fi
+ if test "$ac_cv_lib_png14_png_create_read_struct" = "yes" -a "$ac_cv_lib_z_deflateEnd" = "yes" -a \( ! -e /usr/lib/libpng14.la -o ! -e /usr/lib/libz.la \); then
+  STATIC_LIBS="$STATIC_LIBS -lpng14 -lz"
+  AC_DEFINE(_STATIC_LIBPNG)
+ fi
+ if test "$ac_cv_lib_png15_png_create_read_struct" = "yes" -a "$ac_cv_lib_z_deflateEnd" = "yes" -a \( ! -e /usr/lib/libpng15.la -o ! -e /usr/lib/libz.la \); then
+  STATIC_LIBS="$STATIC_LIBS -lpng15 -lz"
+  AC_DEFINE(_STATIC_LIBPNG)
+ fi
+ if test "$ac_cv_lib_png16_png_create_read_struct" = "yes" -a "$ac_cv_lib_z_deflateEnd" = "yes" -a \( ! -e /usr/lib/libpng16.la -o ! -e /usr/lib/libz.la \); then
+  STATIC_LIBS="$STATIC_LIBS -lpng16 -lz"
+  AC_DEFINE(_STATIC_LIBPNG)
+ fi
  if test "$ac_cv_lib_tiff_TIFFOpen" = "yes" -a ! -e /usr/lib/libtiff.la; then
   STATIC_LIBS="$STATIC_LIBS -ltiff"
   AC_DEFINE(_STATIC_LIBTIFF)
