From f451df702999ed54030c628ad9f50ad0a2cf5ac5 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Wed, 26 Apr 2017 02:24:20 -0400
Subject: [PATCH] tweak systemd unit dir from "/usr/lib" to "/lib"

Upstream-Status: Inappropriate [wr-installer specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 setup.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/setup.py b/setup.py
index cd28118..43cc40d 100644
--- a/setup.py
+++ b/setup.py
@@ -8,7 +8,7 @@ data_files = [('/usr/sbin', ['progs/firstboot']),
               ('/usr/share/firstboot/themes/default',
                glob('themes/default/*.png')),
               ('/usr/share/firstboot/modules', glob('modules/*.py')),
-              ('/usr/lib/systemd/system', glob('systemd/*.service'))]
+              ('/lib/systemd/system', glob('systemd/*.service'))]
 
 # add the firstboot start script for s390 architectures
 if os.uname()[4].startswith('s390'):
-- 
2.8.1

