From 9f516ae49d4ce95dc9054934733ee63eb8ba37bc Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Fri, 19 May 2017 16:22:20 +0800
Subject: [PATCH 15/65] tweak grub config file for WRLinux

Upstream-Status: Inappropriate [oe specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/bootloader.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/pyanaconda/bootloader.py b/pyanaconda/bootloader.py
index a8fc01f..2a91759 100644
--- a/pyanaconda/bootloader.py
+++ b/pyanaconda/bootloader.py
@@ -1506,7 +1506,7 @@ class GRUB2(GRUB):
 
         defaults = open(defaults_file, "w+")
         defaults.write("GRUB_TIMEOUT=%d\n" % self.timeout)
-        defaults.write("GRUB_DISTRIBUTOR=\"$(sed 's, release .*$,,g' /etc/system-release)\"\n")
+        defaults.write("GRUB_DISTRIBUTOR=\"%s\"\n" % productName)
         defaults.write("GRUB_DEFAULT=saved\n")
         defaults.write("GRUB_DISABLE_SUBMENU=true\n")
         if self.console and self.has_serial_console:
-- 
2.7.4

