Index: b/Makefile
===================================================================
--- a/Makefile
+++ b/Makefile
@@ -44,7 +44,7 @@ export bindir      = $(prefix)/bin
 export libdir      = $(prefix)/lib
 export mandir      = $(prefix)/man
 export INSTALLDIR  = $(prefix)/lib/klibc
-export INSTALLROOT =
+export INSTALLROOT = $(INSTALLROOT)
 
 # Create a fake .config as present in the kernel tree
 # But if it exists leave it alone
