---
 Makefile |    4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

Index: b/Makefile
===================================================================
--- a/Makefile
+++ b/Makefile
@@ -1,6 +1,5 @@
-PYVER  := $(shell python -c 'import sys; print sys.version[0:3]')
-PYTHON = python$(PYVER)
-PYTHONINCLUDE = /usr/include/$(PYTHON)
+PYTHON = $(PYTHON_DIR)
+PYTHONINCLUDE = $(STAGING_INCDIR)/$(PYTHON)
 
 VERSION=1.0.7
 
