Index: b/setup.py
===================================================================
--- a/setup.py
+++ b/setup.py
@@ -352,8 +352,9 @@ def main(argv):
             extra_compile_args = debug_compile_args
             argv.remove(arg)
 
-    nss_include_dir  = find_include_dir(['nss3', 'nss'],   ['nss.h',  'pk11pub.h'])
-    nspr_include_dir = find_include_dir(['nspr4', 'nspr'], ['nspr.h', 'prio.h'])
+    base_include_dir = os.getenv("STAGING_INCDIR")
+    nss_include_dir  = find_include_dir(['nss3', 'nss', '.'],   ['nss.h',  'pk11pub.h'], [base_include_dir])
+    nspr_include_dir = find_include_dir(['nspr4', 'nspr', '.'], ['nspr.h', 'prio.h'], [base_include_dir])
 
     nss_error_extension = \
         Extension('nss.error',
