From 9a055c56cf24441129d3d7537608480f2e0203d0 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Thu, 11 Jun 2015 17:44:25 +0800
Subject: [PATCH 02/62] scripts/run-anaconda: replace /usr/bin/bash with
 /bin/sh

Upstream-Status: Inappropriate [wrlinux specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 scripts/run-anaconda | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/scripts/run-anaconda b/scripts/run-anaconda
index d31f70c..142ceb9 100755
--- a/scripts/run-anaconda
+++ b/scripts/run-anaconda
@@ -1,4 +1,4 @@
-#!/usr/bin/bash
+#!/bin/sh
 #
 # Quick script to attach to tmux where anaconda is running
 #
-- 
2.7.4

