From 36bea815e49f21957302aff1a27340bc097ff44b Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Thu, 11 Jun 2015 15:43:20 +0800
Subject: [PATCH 03/62] widgets/Makefile.am: do not compile doc

Upstream-Status: Inappropriate [wrlinux specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 widgets/Makefile.am | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/widgets/Makefile.am b/widgets/Makefile.am
index 7afcf19..be28f54 100644
--- a/widgets/Makefile.am
+++ b/widgets/Makefile.am
@@ -21,7 +21,7 @@
 
 ACLOCAL_AMFLAGS = -I m4
 
-SUBDIRS = src python glade doc data
+SUBDIRS = src python glade data
 
 MAINTAINERCLEANFILES = Makefile.in aclocal.m4 config.guess config.h.in config.sub \
 	configure depcomp install-sh ltmain.sh missing py-compile \
-- 
2.7.4

