From f0e38db01465c865fd3fd842992af7229849efa3 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Sun, 17 Apr 2016 03:58:18 -0400
Subject: [PATCH 11/22] tweak auditd invoking dir

Upstream-Status: Inappropriate [wrlinux specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 anaconda | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/anaconda b/anaconda
index d682a82..b122ca8 100755
--- a/anaconda
+++ b/anaconda
@@ -995,7 +995,7 @@ if __name__ == "__main__":
     if can_touch_runtime_system("start audit daemon"):
         # auditd will turn into a daemon and exit. Ignore startup errors
         try:
-            iutil.execWithRedirect("/sbin/auditd", [])
+            iutil.execWithRedirect("/usr/libexec/anaconda/auditd", [])
         except OSError:
             pass
 
-- 
2.8.1

