From a0b6cdfb9fea0f9c7c7f7fc5d14e26129da2d247 Mon Sep 17 00:00:00 2001
From: Mark Hatle <mark.hatle@windriver.com>
Date: Thu, 5 May 2016 04:02:25 -0400
Subject: [PATCH 16/22] tweak grub config file for WRLinux

Upstream-Status: Inappropriate [wrlinux specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 pyanaconda/bootloader.py | 19 +++++++++++++------
 1 file changed, 13 insertions(+), 6 deletions(-)

diff --git a/pyanaconda/bootloader.py b/pyanaconda/bootloader.py
index efcff4e..d9df6c4 100644
--- a/pyanaconda/bootloader.py
+++ b/pyanaconda/bootloader.py
@@ -1498,14 +1498,21 @@ class GRUB2(GRUB):
 
         defaults = open(defaults_file, "w+")
         defaults.write("GRUB_TIMEOUT=%d\n" % self.timeout)
-        defaults.write("GRUB_DISTRIBUTOR=\"$(sed 's, release .*$,,g' /etc/system-release)\"\n")
+        defaults.write("GRUB_DISTRIBUTOR=\"%s\"\n" % productName)
         defaults.write("GRUB_DEFAULT=saved\n")
         defaults.write("GRUB_DISABLE_SUBMENU=true\n")
-        if self.console and self.has_serial_console:
-            defaults.write("GRUB_TERMINAL=\"serial console\"\n")
-            defaults.write("GRUB_SERIAL_COMMAND=\"%s\"\n" % self.serial_command)
-        else:
-            defaults.write("GRUB_TERMINAL_OUTPUT=\"%s\"\n" % self.terminal_type)
+        # Original
+        #if self.console and self.has_serial_console:
+        #    defaults.write("GRUB_TERMINAL=\"serial console\"\n")
+        #    defaults.write("GRUB_SERIAL_COMMAND=\"%s\"\n" % self.serial_command)
+        #else:
+        #    defaults.write("GRUB_TERMINAL_OUTPUT=\"%s\"\n" % self.terminal_type)
+        # End Original
+        # Force serial and vga
+        defaults.write("GRUB_TERMINAL_OUTPUT=\"console serial\"\n")
+        defaults.write("GRUB_TERMINAL_INPUT=\"console serial\"\n")
+        defaults.write("GRUB_SERIAL_COMMAND=\"serial --unit=0 --speed 115200\"\n")
+        # EndForce serial and vga
 
         # this is going to cause problems for systems containing multiple
         # linux installations or even multiple boot entries with different
-- 
2.8.1

