--- mozilla/security/nss/lib/ssl/sslsock.c	2011-10-22 18:45:40.000000000 +0200
+++ mozilla/security/nss/lib/ssl/sslsock.c.oden	2011-11-05 17:45:55.672091551 +0100
@@ -181,7 +181,7 @@ static sslOptions ssl_defaults = {
     PR_FALSE,   /* noLocks            */
     PR_FALSE,   /* enableSessionTickets */
     PR_FALSE,   /* enableDeflate      */
-    2,          /* enableRenegotiation (default: requires extension) */
+    3,          /* enableRenegotiation (default: transitional) */
     PR_FALSE,   /* requireSafeNegotiation */
     PR_FALSE,   /* enableFalseStart   */
     PR_TRUE     /* cbcRandomIV        */
