From 07c2c267c9cbe7c0368d0b830b9c51b8ba2c2895 Mon Sep 17 00:00:00 2001
From: Hongxu Jia <hongxu.jia@windriver.com>
Date: Fri, 19 May 2017 15:22:19 +0800
Subject: [PATCH 12/62] tweak auditd invoking dir

Upstream-Status: Inappropriate [wrlinux specific]

Signed-off-by: Hongxu Jia <hongxu.jia@windriver.com>
---
 anaconda.py | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/anaconda.py b/anaconda.py
index 80772b6..5f4f74e 100755
--- a/anaconda.py
+++ b/anaconda.py
@@ -470,7 +470,7 @@ if __name__ == "__main__":
     if can_touch_runtime_system("start audit daemon"):
         # auditd will turn into a daemon and exit. Ignore startup errors
         try:
-            iutil.execWithRedirect("/sbin/auditd", [])
+            iutil.execWithRedirect("/usr/libexec/anaconda/auditd", [])
         except OSError:
             pass
 
-- 
2.7.4

