tigervnc: disable vncview

Upstream-Status: Pending

Signed-off-by: Chong Lu <Chong.Lu@windriver.com>
---
 CMakeLists.txt | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/CMakeLists.txt b/CMakeLists.txt
index ce0a00a..765878a 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -404,7 +404,7 @@ endif()
 
 option(BUILD_VIEWER "Build TigerVNC viewer" ON)
 if(BUILD_VIEWER)
-  add_subdirectory(vncviewer)
+#  add_subdirectory(vncviewer)
   add_subdirectory(media)
 endif()
 
-- 
1.9.1

