--- anaconda-16.25/pyanaconda/packages.py.orig	2014-06-09 15:09:53.177393790 +0800
+++ anaconda-16.25/pyanaconda/packages.py	2014-06-09 15:10:39.978717748 +0800
@@ -299,8 +299,7 @@
         ts.dbMatch('provides', 'plymouth').count()):
         anaconda.bootloader.boot_args.update(["rhgb", "quiet"])
 
-    if ts.dbMatch('provides', 'service(graphical-login)').count() and \
-       ts.dbMatch('provides', 'xorg-x11-server-Xorg').count() and \
+    if ts.dbMatch('provides', 'xserver-xorg').count() and \
        anaconda.displayMode == 'g' and not flags.usevnc:
         anaconda.desktop.setDefaultRunLevel(5)
 
