Disable the language/i18n configuration

Temporarily disable the languages/i18n configuration.

Signed-off-by: Mark Hatle <mark.hatle@windriver.com>

diff -ur pyanaconda/language.py /home/mhatle/anaconda-WIP/pyanaconda/language.py
--- a/pyanaconda/language.py	2011-11-02 18:49:54.000000000 -0500
+++ b/pyanaconda/language.py	2014-01-18 23:40:19.929136102 -0600
@@ -286,6 +286,8 @@
         return self.nativeLangNames[lang]
 
     def write(self, instPath):
+        return
+
         f = open(instPath + "/etc/sysconfig/i18n", "w")
 
         for (key, val) in self.info.iteritems():
