Index: anaconda-16.25/pyanaconda/backend_log.py
===================================================================
--- anaconda-16.25.orig/pyanaconda/backend_log.py
+++ anaconda-16.25/pyanaconda/backend_log.py
@@ -27,7 +27,7 @@ import signal
 import anaconda_log
 import iutil
 
-SYSLOG_PATH           = '/sbin/rsyslogd'
+SYSLOG_PATH           = '/usr/sbin/rsyslogd'
 SYSLOG_PIDFILE        = '/var/run/rsyslog_backend.pid'
 SYSLOG_CFGFILE        = '/tmp/rsyslog_backend.conf'
 
