From f354c85e0577b3eea84b4b06ba303086e95362ab Mon Sep 17 00:00:00 2001
Message-Id: <f354c85e0577b3eea84b4b06ba303086e95362ab.1367526210.git.Jim.Somerville@windriver.com>
From: Jim Somerville <Jim.Somerville@windriver.com>
Date: Thu, 2 May 2013 16:22:38 -0400
Subject: [PATCH 1/1] Allow install to usr/lib instead of usr/local/lib

Signed-off-by: Jim Somerville <Jim.Somerville@windriver.com>
---
 Makefile |    2 +-
 1 files changed, 1 insertions(+), 1 deletions(-)

diff --git a/Makefile b/Makefile
index 23b98c2..fec8a67 100644
--- a/Makefile
+++ b/Makefile
@@ -24,7 +24,7 @@ test:
 	PYTHONPATH=. python $(TESTSUITE) -v
 
 install: all
-	python setup.py install --root=$(DESTDIR)
+	python setup.py install --root=$(DESTDIR) --prefix=/usr
 	$(MAKE) -C po install
 
 ChangeLog:
-- 
1.7.4.1

